example/lib/data/app_state.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:flutter/foundation.dart';
+import 'package:scoped_model/scoped_model.dart';
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:airship_example/screens/message_view.dart';
+
+import 'package:airship/airship.dart';
+class AppState extends Model {
+  Future<String> channelId = Airship.channelId;
+  Future<String> namedUser = Airship.namedUser;
+  Future<bool> notificationsEnabled = Airship.userNotificationsEnabled;
+  Future<List<String>> tags = Airship.tags;
+  Future<List<InboxMessage>> messages = Airship.inboxMessages;
+
+  void setUserNotificationsEnabled(bool enabled) {
+    Airship.setUserNotificationsEnabled(enabled);
+    Airship.userNotificationsEnabled.then((isEnabled) {
+      notificationsEnabled = Future(() {
```

Comment:
I'm not too familiar with dart, or flutter for that matter, so this may be a dumb question. But why are we redefining notificationsEnabled here?